jobs/model_edit.rb

```
@@ -0,0 +1,68 @@
+require 'resque'
+require 'resque-status'
+require 'sinatra/activerecord'
+require 'redis'
+require_relative './reload_cache.rb'
+require_relative '../models/videos'
+require_relative '../models/articles'
+require_relative '../models/feeds'
+require_relative '../models/resources'
+
+module Jobs
+    module Models
+        class Edit
+            include Resque::Plugins::Status
+            
+            #@queue = :models
+            
+            def perform
+                # Find object in database.
+                begin
+                    @type = options['model_type'].split('::').reduce(Module, :const_get)
+                    @obj = @type.find(options['model_id'])
+                rescue ActiveRecord::RecordNotFound
+                    # Invalid ID.
+                    failed(msg: 'Model not found')
+                    return
+                rescue NameError
+                    # Invalid model type.
+                    failed(msg: 'Invalid model ref')
+                    return
+                rescue NoMethodError
+                    # Invalid model type.
+                    failed(msg: 'Internal error')
+                    return
+                end
+                # Update progress.
+                at(1, 4)
+                # Verify permissions.
+                if options['user_id'] == @obj.author || options['user_id'] == 'super' || options['is_super']
+                    # Update progress.
+                    at(2,4)
+                    # Update the object.
+                    begin
+                        @obj.update!(options['args'])
+                    rescue ActiveRecord::RecordInvalid
+                        failed(msg: 'Invalid parameters')
+                        return
+                    rescue
```

Shouldn't be suppressing exceptions with generic catch statement.